crates/rethnet_evm/benches/util.rs

```diff
@@ -86,61 +129,79 @@ mod config {
     ];
 
     pub const STORAGE_SCALES: [u64; 4] = [1, 10, 100, 1000];
+
+    pub const SNAPSHOT_SCALES: [u64; 4] = [1, 10, 100, 1000];
```

Comment:
`CHECKPOINT_SCALES` needs to be dividable without remainder by every scale of  `SNAPSHOT_SCALES` or the benchmark will end up having fluctuations in number of accounts, etc. due to rounding.

To do so,

```rust
pub const SNAPSHOT_SCALES: [u64; NUM_SCALES] = [1, 5, 10, 20];
const MAX_SNAPSHOT_SCALE: u64 = SNAPSHOT_SCALES[NUM_SCALES - 1];

pub const CHECKPOINT_SCALES: [u64; NUM_SCALES] = [
  MAX_SNAPSHOT_SCALE,
  MAX_SNAPSHOT_SCALE * 2
  MAX_SNAPSHOT_SCALE * 4,
  MAX_SNAPSHOT_SCALE * 8
];
const MAX_CHECKPOINT_SCALE: u64 = CHECKPOINT_SCALES[NUM_SCALES - 1];

pub const ADDRESS_SCALES: [u64; NUM_SCALES] = [
  MAX_CHECKPOINT_SCALE,
  MAX_CHECKPOINT_SCALE * 5,
  MAX_CHECKPOINT_SCALE * 25,
  MAX_CHECKPOINT_SCALE * 50,
];
```

What that means is that we vary:

- no. of snapshots: 1, 5, 10, 20
- no. of checkpoints: 20, 40, 80, 160
- no. of accounts: 160, 800, 4000, 8000

(no. storage is not taken into consideration as it's unrelated

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3921/commits/a59897b66f25e6c4bd0e249afc1087ca84fef3e9